packages/cli/src/commands/rewards/show.ts

```diff
@@ -0,0 +1,293 @@
+    ...BaseCommand.flags,
+    voter: Flags.address({ description: 'Voter to show rewards for' }),
+    validator: Flags.address({ description: 'Validator to show rewards for' }),
+    group: Flags.address({ description: 'Validator Group to show rewards for' }),
+    epochs: flags.integer({
+      default: 1,
+      description: 'Show results for the last N epochs',
+    }),
+  }
+
+  static args = []
+
+  static examples = ['show --address 0x5409ed021d9299bf6814279a6a1411a7e866a631']
+
+  async run() {
+    const res = this.parse(Show)
+    const election = await this.kit.contracts.getElection()
+    const validators = await this.kit.contracts.getValidators()
+    const epochSize = await validators.getEpochSize()
+    const currentBlock = await this.web3.eth.getBlockNumber()
+    const lastEpochBlock = Math.floor(currentBlock / epochSize) * epochSize
+    const fromBlock: number = lastEpochBlock - epochSize * ((res.flags.epochs || 1) - 1)
+    let voterRewards: VoterReward[] = []
+    let validatorRewards: ValidatorReward[] = []
+    let validatorGroupRewards: ValidatorGroupReward[] = []
+    let address = ''
+
+    if (res.flags.voter) {
+      address = res.flags.voter
+      await newCheckBuilder(this)
+        .isAccount(address)
+        .runChecks()
+    }
+
+    if (res.flags.validator) {
+      if (address) throw Error('Must select --voter, --validator, or --group')
+      address = res.flags.validator
+      await newCheckBuilder(this)
+        .isValidator(address)
+        .runChecks()
+    }
+
+    if (res.flags.group) {
+      if (address) throw Error('Must select --voter, --validator, or --group')
+      address = res.flags.group
+      await newCheckBuilder(this)
+        .isValidatorGroup(address)
+        .runChecks()
+    }
+
+    // For each epoch...
+    for (let blockNumber = fromBlock; blockNumber <= lastEpochBlock; blockNumber += epochSize) {
+      // Get the groups that address voted for at this epoch, and those groups' total votes.
+      const activeAddressVotes: { [key: string]: BigNumber } = {}
+      const activeGroupVotes: { [key: string]: BigNumber } = {}
+      if (address) {
+        const voter = await election.getVoter(address, blockNumber)
+        for (const vote of voter.votes) {
+          const group: string = vote.group.toLowerCase()
+          activeAddressVotes[group] = (activeAddressVotes[group] || new BigNumber(0)).plus(
+            vote.active
+          )
+          activeGroupVotes[group] = (activeGroupVotes[group] || new BigNumber(0)).plus(
+            await election.getTotalVotesForGroup(group, blockNumber)
+          )
+        }
+      }
+
+      // voterRewardsEvent applies to address when voterRewardsEvent.group in activeAddressVotes.
+      const epochVoterRewardsEvents = await election.getPastVoterRewards({
+        fromBlock: blockNumber,
+        toBlock: blockNumber,
+      })
+
+      const voterRewardsValidatorGroupDetails: ValidatorGroup[] = await concurrentMap(
```

Comment:
I think that instead of returning the generic `EventLog` you can return something called `VoterRewards` which contains the information already digested